src/mcts/stoppers/alphazero.cc

```diff
@@ -36,33 +36,49 @@ class AlphazeroTimeManager : public TimeManager {
   AlphazeroTimeManager(int64_t move_overhead, const OptionsDict& params)
       : move_overhead_(move_overhead),
         alphazerotimepct_(
-            params.GetOrDefault<float>("alphazero-time-pct", 12.0f)) {
+            params.GetOrDefault<float>("alphazero-time-pct", 12.0f)),
+        alphazeroincrementpct_(
+            params.GetOrDefault<float>("alphazero-increment-pct", 95.0f)) {
     if (alphazerotimepct_ < 0.0f || alphazerotimepct_ > 100.0f)
       throw Exception("alphazero-time-pct value to be in range [0.0, 100.0]");
+    if (alphazeroincrementpct_ < 0.0f || alphazeroincrementpct_ > 100.0f)
+      throw Exception(
+          "alphazero-increment-pct value to be in range [0.0, 100.0]");
   }
   std::unique_ptr<SearchStopper> GetStopper(const GoParams& params,
                                             const NodeTree& tree) override;
 
  private:
   const int64_t move_overhead_;
   const float alphazerotimepct_;
+  const float alphazeroincrementpct_;
 };
 
 std::unique_ptr<SearchStopper> AlphazeroTimeManager::GetStopper(
     const GoParams& params, const NodeTree& tree) {
   const Position& position = tree.HeadPosition();
   const bool is_black = position.IsBlackToMove();
   const std::optional<int64_t>& time = (is_black ? params.btime : params.wtime);
+  const std::optional<int64_t>& inc = (is_black ? params.binc : params.winc);
+  const int increment = inc ? std::max(int64_t(0), *inc) : 0;
   // If no time limit is given, don't stop on this condition.
   if (params.infinite || params.ponder || !time) return nullptr;
 
   auto total_moves_time = *time - move_overhead_;
 
-  float this_move_time = total_moves_time * (alphazerotimepct_ / 100.0f);
+  float this_move_time = std::max<int64_t>(0, total_moves_time - increment) *
+      (alphazerotimepct_ / 100.0f) + increment *
+      (alphazeroincrementpct_ / 100.0f);
 
-  LOGFILE << "Budgeted time for the move: " << this_move_time << "ms"
-          << "Remaining time " << *time << "ms(-" << move_overhead_
-          << "ms overhead)";
+  // If increment time is added only after a move has been made,
```

Comment:
Can it really happen? It's `total_move_time - increment` above.

Reply:
Only in the first move.
In official FIDE rules it says: `increment: 6.1. An amount of time (from 2 to 60 seconds) added from the start before each move for the player.` Also in blitz games of world class players you can see 03:02 on the clock before the start of the game with 3+2 time control.
But here some say it should be added after a move: https://old.chesstempo.com/chess-forum/general_chess_discussion/question_about_fisher_clock-t2042.0.html Maybe it's the old FIDE rules.

Reply:
Indeed, I didn't think about the first move.
I think in human (and WCCC) championships it's indeed added before the first move, but in computer ones it's after.

It can also happen in werd TCs like "1 hour for first 40 moves, than 15 seconds increment per move".